src/transaction/log_replication_mvcc.cpp

```diff
@@ -37,13 +38,75 @@ namespace cublog
     const auto found_it = m_mapped_mvccids.find (tranid);
     if (found_it == m_mapped_mvccids.cend ())
       {
-	m_mapped_mvccids.emplace (tranid, mvccid);
+	m_mapped_mvccids.emplace (tranid, tran_mvccid_info { mvccid });
       }
     else
       {
 	// only one mvccid per transaction is assumed
 	// sub-transaction mvccid's are not implemented yet
-	assert (found_it->second == mvccid);
+	assert (found_it->second.id == mvccid);
+      }
+
+    if (prm_get_bool_value (PRM_ID_ER_LOG_PTS_REPL_DEBUG))
+      {
+	_er_log_debug (ARG_FILE_LINE, "[REPLICATOR_MVCC] new_assigned_mvccid tranid=%d mvccid=%llu\n",
+		       tranid, (unsigned long long)mvccid);
+	dump_map ();
+      }
+  }
+
+  void
+  replicator_mvcc::new_assigned_mvccid (TRANID tranid, MVCCID mvccid, MVCCID parent_mvccid)
+  {
+    assert (MVCCID_IS_NORMAL (mvccid));
+
+    if (MVCCID_IS_VALID (parent_mvccid))
+      {
+	// mvccid is a sub-id, as it has a valid parent mvccid
+	assert (MVCCID_IS_NORMAL (parent_mvccid));
+
+	const auto found_it = m_mapped_mvccids.find (tranid);
+
+	//assert (found_it == m_mapped_mvccids.cend ());
+	if (found_it == m_mapped_mvccids.cend ())
+	  {
+	    if (prm_get_bool_value (PRM_ID_ER_LOG_PTS_REPL_DEBUG))
+	      {
+		_er_log_debug (ARG_FILE_LINE, "[REPLICATOR_MVCC] new_assigned_sub_mvccid"
+			       " WARNING sudden new parent_mvccid tranid=%d parent_mvccid=%llu\n",
+			       tranid, (unsigned long long)parent_mvccid);
+		dump_map ();
+	      }
+
+	    // TODO: a new parent mvccid appeared out of the blue;
+	    // not sure whether this is a valid scenario
+	    // however, it will be commited/aborted at the end of the transaction
+	    new_assigned_mvccid (tranid, parent_mvccid);
+	  }
+
+	if (found_it != m_mapped_mvccids.cend ())
+	  {
+	    assert (found_it->second.id == mvccid);
```

Comment:
Shouldn't this be  ?
```suggestion
	    assert (found_it->second.id == parent_mvccid);
```

Reply:
Corrected